src/model/SortByPublicationDate.java

```diff
@@ -1,11 +1,53 @@
 package model;
+import java.util.*;
+
+/**
+ * @Author Joe and Peyton
+ */
 
 public class SortByPublicationDate implements SortStrategy{
 
+    private PersonalCollection personalCollection;
+
+    public SortByPublicationDate(PersonalCollection pc){
+        this.personalCollection = pc;
+    }
+
     @Override
-    public PersonalCollection algorithm() {
-        // TODO Auto-generated method stub
-        throw new UnsupportedOperationException("Unimplemented method 'algorithm'");
+    public ArrayList<ComicBook> algorithm(String toBeSorted, boolean isSortDb) throws Exception {
```

Comment:
THIS GOES FOR ALL OF THE SORTS THAT YOU HAVE
this should instead be returning a PersonalCollection since its setting the personal collection with a new sorted arraylist